lib/CodeGen/CGExprScalar.cpp

```diff
@@ -1444,6 +1444,10 @@ Value *ScalarExprEmitter::VisitCastExpr(CastExpr *CE) {
     return CGF.EmitDynamicCast(V, DCE);
   }
 
+  case CK_DynamicPtrBounds:
+  case CK_AssumePtrBounds:
+    return CGF.EmitDynamicBoundsCast(CE);
```

Comment:
EmitDynamicBoundsCast has two functionalities
First, it is code generation for explicit type casting because bounds_cast operation is also explicit type casting operatoin
Second, it emits dynamic_checks for dynamic_bounds_cast only
In EmitDynamicBoundsCast method, it generats casting operation for both bounds cast type
and it only generates bounds operation for dynamic_bounds_cast operation